internal/convert/registryv1.go

```diff
@@ -279,13 +276,19 @@ func Convert(in RegistryV1, installNamespace string, targetNamespaces []string)
 
 	for _, permission := range permissions {
 		saName := saNameOrDefault(permission.ServiceAccountName)
-		name := generateName(fmt.Sprintf("%s-%s", in.CSV.Name, saName), []interface{}{in.CSV.Name, permission})
+		name, err := generateName(fmt.Sprintf("%s-%s", in.CSV.Name, saName), []interface{}{in.CSV.Name, permission})
+		if err != nil {
+			return nil, err
+		}
 		roles = append(roles, newRole(installNamespace, name, permission.Rules))
 		roleBindings = append(roleBindings, newRoleBinding(installNamespace, name, name, installNamespace, saName))
 	}
 	for _, permission := range clusterPermissions {
 		saName := saNameOrDefault(permission.ServiceAccountName)
-		name := generateName(fmt.Sprintf("%s-%s", in.CSV.Name, saName), []interface{}{in.CSV.GetName(), permission})
+		name, err := generateName(fmt.Sprintf("%s-%s", in.CSV.Name, saName), []interface{}{in.CSV.GetName(), permission})
```

Comment:
Use an anonymous struct here to make it JSON-serializable (see my changes in v0).